### DISCUSSION TITLE
How to avoid need for model_dump(by_alias=True)? Not "alias" but "rename"

### DISCUSSION THREAD

**Discussion by alexanderankin** [pydantic/pydantic#8725] [Question]
consider an example input data:

```json
{
  "message": "message is voided"
  "void": true
}
```

with java's jackson, you'd write `@lombok.Data class Message { String message; @JsonProperty("void") String void_; }` and proceed as normal. They support aliases with `@JsonAlias`, as it is a separate thing.

with pydantic, it seems you are forced into polluting your serialization code with `by_alias=True`, perhaps because pydantic is forcing you to use fields with unreserved names, which seems needlessly inflexible.

```python
class Message(BaseModel):
  message: str
  void_: str = Field(alias='void')

print(Message(**json.loads('{"message":"message is voided", "void":true}'))) # fine
print(Message(**json.loads('{"message":"message is voided", "void":true}')).model_dump()) # void_!!!
print(Message(**json.loads('{"message":"message is voided", "void":true}')).model_dump(by_alias=True)) # !?!?!
```

**Comment by alexanderankin** [comment]
linking this here https://github.com/pydantic/pydantic/issues/8551#issuecomment-1926005882
https://github.com/pydantic/pydantic/issues/8379
https://github.com/pydantic/pydantic/pull/9004/files

https://github.com/pydantic/pydantic/discussions/4456

**Comment by dohrenstein** [comment]
I have a model which needs a field called "class" and am running into the same requirement (to use model_dump(by_alias=True) everywhere)

**Comment by CharlesPerrotMinot** [comment]
It's pretty easy to override it though.
Create your own base class, and have your own dump_json like so
```
class Model(BaseModel):
    def model_dump(
        self,
        by_alias: bool = True,
        **kwargs: Any,
    ) -> dict[str, Any]:
        return super().model_dump(by_alias=by_alias, **kwargs)
```

**Reply by CharlesPerrotMinot** [reply]
That being said it could be nice to have a direct option for that in the ConfigDict @samuelcolvin

**Reply by alexisdrakopoulos** [reply]
Agreed would be nice to add to the config

**Reply by CharlesPerrotMinot** [reply]
Any chance @sydney-runkle?

**Reply by alexanderankin** [reply]
its being worked on but its hard because you have to write rust

**Reply by alexanderankin** [reply]
@CharlesPerrotMinotHCHB https://github.com/pydantic/pydantic/issues/8379#issuecomment-2581063061
